Makefile

```
@@ -13,111 +15,104 @@ L=[01;30m
-
-_empaquetar_osx:
-	rm -f binarios/${NOMBRE}-${VERSION}.dmg
-	$(call empaquetar,darwin,all,icns)
-	hdiutil create binarios/${NOMBRE}-${VERSION}.dmg -srcfolder ./binarios/${NOMBRE}-darwin-x64/${NOMBRE}.app -size 1g
+build_live:
+	$(EMBER) build --watch
 
-_empaquetar_win32:
-	$(call empaquetar,win32,ia32,ico)
-	@echo "${G}Generando instalador para windows...${N}"
-	cp packaging/instalador.nsi binarios/${NOMBRE}-win32-ia32/
-	cd binarios/${NOMBRE}-win32-ia32/; makensis instalador.nsi
-	@mv binarios/${NOMBRE}-win32-ia32/${NOMBRE}.exe binarios/${NOMBRE}-${VERSION}.exe
+release_patch:
+	$(EMBER) release
 
-_empaquetar_linux: _borrar_binarios_linux _empaquetar_zip_linux_x64 _empaquetar_zip_linux_ia32 _empaquetar_deb_linux_x64
+release_minor:
+	$(EMBER) release --minor
 
-_borrar_binarios_linux:
-	rm -rf binarios/${NOMBRE}-*linux*
-	rm -rf binarios/*.deb
+release_major:
+	$(EMBER) release --major
 
-# Este empaquetado tiene el problema de que NO reemplaza la aplicación vieja de Huayra.
-# Además, el package debian generado tiene nombre diferente al viejo.
-_empaquetar_deb_linux_x64:
-	$(call empaquetar,linux,x64,icns)
+package = @echo "${G}Generating package for $(1) $(2)...${N}"; node_modules/.bin/electron-packager dist ${NAME} --app-version=${VERSION} --platform=$(1) --arch=$(2) --ignore=node_modules --out=binarios --overwrite --icon=packaging/icono.$(3)
+
+electron_pre_build:
+	@ echo "${G}Making dist directory works with electron...${N}"
+	if ! ([ -d "./binaries" ]); then mkdir ./binaries; fi
+	cp package.json dist/package.json
+	cp packaging/electron.js dist
+
+electron_osx_package:
+	make electron_pre_build;
+	@ echo "${G}Generating package for osx...${N}"
+	rm -f binarios/${NAME}-${VERSION}.dmg
+	$(call package, darwin, all, icns)
+	hdiutil create binarios/${NAME}-${VERSION}.dmg -srcfolder ./binarios/${NAME}-darwin-x64/${NAME}.app -size 1g
+
+electron_win32_package:
+	make electron_pre_build;
+	@ echo "${G}Generating installer for windows package...${N}"
+	$(call package, win32, ia32, ico)
+	cp packaging/instalador.nsi binarios/${NAME}-win32-ia32/
+	cd binarios/${NAME}-win32-ia32/; makensis instalador.nsi
+	mv binarios/${NAME}-win32-ia32/${NAME}.exe binarios/${NAME}-${VERSION}.exe
+
+electron_linux_packages: 
+	make electron_linux_x64_zip_package 
+	make electron_linux_ia32_zip_package 
+	make electron_linux_x64_deb_package
+
+# This packaging has the problem that it does NOT replace the old Huayra Pilas Bloques app.
+# Also, the debian generator package has a different name than the old one
+electron_linux_x64_deb_package:
+	make electron_pre_build;
+	@ echo "${G}Generating linux x64 package for debian...${N}"
+	if [ -f "./binaries/${NAME}_${VERSION}_amd64.deb" ]; then rm "./binaries/${NAME}_${VERSION}_amd64.deb"; fi
```

I'd use `rm -f` everytime we need something like this. The if is hideous :laughing: 

```suggestion
	rm -f "./binaries/${NAME}_${VERSION}_amd64.deb"
```